module/zfs/vdev_raidz_avx128.c

```diff
@@ -0,0 +1,360 @@
+	asm volatile("vpaddb %xmm13,%xmm13,%xmm13");			\
+	asm volatile("vpand %xmm3,%xmm2,%xmm2");			\
+	asm volatile("vpand %xmm3,%xmm6,%xmm6");			\
+	asm volatile("vpand %xmm3,%xmm10,%xmm10");			\
+	asm volatile("vpand %xmm3,%xmm14,%xmm14");			\
+	asm volatile("vpxor %xmm2,%xmm1,%xmm1");			\
+	asm volatile("vpxor %xmm6,%xmm5,%xmm5");			\
+	asm volatile("vpxor %xmm10,%xmm9,%xmm9");			\
+	asm volatile("vpxor %xmm14,%xmm13,%xmm13");			\
+	asm volatile("vpxor %xmm0,%xmm1,%xmm1");			\
+	asm volatile("vpxor %xmm4,%xmm5,%xmm5");			\
+	asm volatile("vpxor %xmm8,%xmm9,%xmm9");			\
+	asm volatile("vpxor %xmm12,%xmm13,%xmm13");			\
+	asm volatile("vmovdqa %%xmm1,%0" : "=m" (*(q+0)));		\
+	asm volatile("vmovdqa %%xmm5,%0" : "=m" (*(q+2)));		\
+	asm volatile("vmovdqa %%xmm9,%0" : "=m" (*(q+4)));		\
+	asm volatile("vmovdqa %%xmm13,%0" : "=m" (*(q+6)))
+
+#define	COMPUTE8_R_AVX128						\
+	asm volatile("vmovdqa %0,%%xmm1" : : "m" (*(r+0)));		\
+	asm volatile("vmovdqa %0,%%xmm5" : : "m" (*(r+2)));		\
+	asm volatile("vmovdqa %0,%%xmm9" : : "m" (*(r+4)));		\
+	asm volatile("vmovdqa %0,%%xmm13" : : "m" (*(r+6)));		\
+	MAKE_CST32_AVX128(xmm3, 0x1d1d1d1d);				\
+	for (j = 0; j < 2; j++) {					\
+		asm volatile("vpxor %xmm14, %xmm14, %xmm14");		\
+		asm volatile("vpcmpgtb %xmm1, %xmm14, %xmm2");		\
+		asm volatile("vpcmpgtb %xmm5, %xmm14, %xmm6");		\
+		asm volatile("vpcmpgtb %xmm9, %xmm14, %xmm10");		\
+		asm volatile("vpcmpgtb %xmm13, %xmm14, %xmm14");	\
+		asm volatile("vpaddb %xmm1,%xmm1,%xmm1");		\
+		asm volatile("vpaddb %xmm5,%xmm5,%xmm5");		\
+		asm volatile("vpaddb %xmm9,%xmm9,%xmm9");		\
+		asm volatile("vpaddb %xmm13,%xmm13,%xmm13");		\
+		asm volatile("vpand %xmm3,%xmm2,%xmm2");		\
+		asm volatile("vpand %xmm3,%xmm6,%xmm6");		\
+		asm volatile("vpand %xmm3,%xmm10,%xmm10");		\
+		asm volatile("vpand %xmm3,%xmm14,%xmm14");		\
+		asm volatile("vpxor %xmm2,%xmm1,%xmm1");		\
+		asm volatile("vpxor %xmm6,%xmm5,%xmm5");		\
+		asm volatile("vpxor %xmm10,%xmm9,%xmm9");		\
+		asm volatile("vpxor %xmm14,%xmm13,%xmm13");		\
+	}								\
+	asm volatile("vpxor %xmm0,%xmm1,%xmm1");			\
+	asm volatile("vpxor %xmm4,%xmm5,%xmm5");			\
+	asm volatile("vpxor %xmm8,%xmm9,%xmm9");			\
+	asm volatile("vpxor %xmm12,%xmm13,%xmm13");			\
+	asm volatile("vmovdqa %%xmm1,%0" : "=m" (*(r+0)));		\
+	asm volatile("vmovdqa %%xmm5,%0" : "=m" (*(r+2)));		\
+	asm volatile("vmovdqa %%xmm9,%0" : "=m" (*(r+4)));		\
+	asm volatile("vmovdqa %%xmm13,%0" : "=m" (*(r+6)))
+
+void
+vdev_raidz_generate_parity_p_avx128(raidz_map_t *rm)
+{
+	uint64_t *p, *src, pcount, ccount, i;
+	int c;
+
+	pcount = rm->rm_col[VDEV_RAIDZ_P].rc_size / sizeof (src[0]);
+	kfpu_begin();
+	for (c = rm->rm_firstdatacol; c < rm->rm_cols; c++) {
+		src = rm->rm_col[c].rc_data;
+		p = rm->rm_col[VDEV_RAIDZ_P].rc_data;
+		ccount = rm->rm_col[c].rc_size / sizeof (src[0]);
+
+		if (c == rm->rm_firstdatacol) {
+			ASSERT(ccount == pcount);
+			i = 0;
+			if (ccount > 7) /* ccount is unsigned */
```

Comment:
There's no need to make it signed and there's no need to use if.
Just a simple `for (i = 0; i + 8 < ccount; i += 8)` will do.